tools/build-mingw64.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+
+# Script to build `ocamlformat' under Windows, using the `mingw64' toolchain.
+# All it requires is a standard Cygwin installation with the `mingw64'
+# toolchain.
+
+# The script builds `flexdll', `ocaml', `dune', `ocamlformat' and their
+# dependencies and leaves the resulting binary at
+#
+#   _build/install/default/bin/ocamlformat.exe
+
+set -euo pipefail
+
+root=$(git rev-parse --show-toplevel)
+
+flexdll_version=0.39
+flexdll_url=https://github.com/alainfrisch/flexdll
+
+ocaml_version=4.12.0
+ocaml_url=https://github.com/ocaml/ocaml
+
+dune_version=2.9.0
+dune_url=https://github.com/ocaml/dune
+
+packages=(
+    https://github.com/janestreet/sexplib0                v0.14.0
+    https://github.com/ocaml-dune/csexp                   master
```

Comment:
we're not terribly reproducible because of transitive deps, but I think we should avoid refering to `master` if possible